packages/material-ui-styles/src/index.d.ts

```diff
@@ -199,15 +198,12 @@ declare module '@material-ui/styles/StylesProvider' {
 }
 
 declare module '@material-ui/styles/ThemeProvider' {
-  import { Theme } from '@material-ui/core';
-
-  const ThemeContext: React.Context<Theme>;
-
-  export interface ThemeProviderProps {
+  export interface ThemeProviderProps<Theme> {
     children: React.ReactNode;
     theme: Theme | ((outerTheme: Theme) => Theme);
   }
-  const ThemeProvider: React.ComponentType<ThemeProviderProps>;
+  type ThemeProvider<T> = React.ComponentType<ThemeProviderProps<T>>;
+  const ThemeProvider: ThemeProvider<any>;
```

Comment:
I think this looses the generic type info. We need to export it as a function that is recognized as a function component. Could you add a test with a theme that is not assignable to the core theme?